components/findMatch.tsx

```diff
@@ -0,0 +1,112 @@
+import React, {Component} from "react";
+import { Button, TextField, ToggleButton, ToggleButtonGroup } from "@mui/material";
+
+export class FindMatch extends Component {
+
+    selected : boolean = false;
+    id : string = "";
+
+    changeStatus = () => {
```

Comment:
tutte le funzioni per motivi di efficienza dovrebbero essere dei [`useCallback`](https://it.reactjs.org/docs/hooks-reference.html)